datetimes/rfc822: support non-standard format

urain39 opened the pull request:
```
Example: 'Fri,24 Nov 2023 18:28:36 -0000'
         ~~~~^~~ No space
```

Rongronggg9 commented on feedparser/datetimes/rfc822.py:
```diff
@@ -68,6 +69,7 @@
     "nov": 11,
     "dec": 12,
 }
+re_separator = re.compile(r"\s+|,\s*")
```
It makes the splitting 3-10x slower.

```console
$ python3 -m timeit -s 'import re; r=re.compile(r"\s+|,\s*"); s=" "*100' 's.split()'
5000000 loops, best of 5: 73.5 nsec per loop
$ python3 -m timeit -s 'import re; r=re.compile(r"\s+|,\s*"); s=" "*100' 'r.split(s)'
1000000 loops, best of 5: 250 nsec per loop
$ python3 -m timeit -s 'import re; r=re.compile(r"\s+|,\s*"); s=", "*100' 'r.split(s)'
50000 loops, best of 5: 7.88 usec per loop
$ python3 -m timeit -s 'import re; r=re.compile(r"\s+|,\s*"); s=", "*100' 's.split()'
500000 loops, best of 5: 696 nsec per loop
```

Rongronggg9 reviewed (commented):
https://github.com/kurtmckee/feedparser/blob/6cbbcdc6b2db8ad35cf969bfe1bd2d9938b1aa2f/feedparser/datetimes/rfc822.py#L92-L94

I would suggest doing the trick here...

urain39 commented:
In my tests, use `re.split` is about 15% slower than use `str.split` in 10000 times.

```py
import time
import rfc822
import rfc822_re


date = 'Fri,24 Nov 2023 18:28:36 -0000'


def timeit(fn):
  begin = time.time()
  for _ in range(10000):
    fn(date)
  return time.time() - begin


print(f'rfc822: {timeit(rfc822._parse_date_rfc822)}')
print(f'rfc822_re: {timeit(rfc822_re._parse_date_rfc822)}')
```

```
rfc822: 0.48220133781433105
rfc822_re: 0.5581390857696533
```

kurtmckee commented on tests/test_date_parsers.py:
```diff
@@ -199,6 +199,12 @@ def test_year_10000_date():
             "Thu Aug 30 2012 17:26:16 +0200",
             (2012, 8, 30, 15, 26, 16, 3, 243, 0),
         ),
+        # Comma without spaces
+        (
+            _parse_date_rfc822,
+            "Fri,24 Nov 2023 18:28:36 -0000",
+            (2023, 11, 24, 18, 28, 36, 4, 328, 0),
+        ),
```
Excellent. Thanks for this test case!

Rongronggg9 commented on feedparser/datetimes/rfc822.py:
```diff
@@ -91,6 +91,10 @@ def _parse_date_rfc822(date):
+        # Comma without spaces:
+        # 'Fri,24 Nov 2023 18:28:36 -0000'
+        if "," in parts[0] and parts[0][-1] != ",":
+            parts.insert(1, parts[0].split(",")[1])
```
Considering the case of double commas, the behavior of `if "," in parts[0] and parts[0][-1] != ",":` maybe undesired when the date is `Fri,, 24 Nov 2023 18:28:36 -0000` (there is a space after the double commas).

So it may be better to do `rpartition()` first:
```python
after_comma = parts[0].rpartition(",")[2]
if after_comma:
    parts.insert(1, after_comma)
```

kurtmckee commented:
@urain39 Thank you very much for this contribution!

@Rongronggg9 Thank you for reviewing this PR!

kurtmckee commented:
#420 made CI more resilient by using better cache-busting, and is significantly faster.
